src/models/network/afi.rs

```diff
@@ -1,8 +1,11 @@
+use num_enum::{IntoPrimitive, TryFromPrimitive};
+
 /// AFI -- Address Family Identifier
 ///
 /// <https://www.iana.org/assignments/address-family-numbers/address-family-numbers.xhtml>
-#[derive(Debug, PartialEq, Primitive, Clone, Copy, Eq, Hash)]
+#[derive(Debug, PartialEq, TryFromPrimitive, IntoPrimitive, Clone, Copy, Eq, Hash)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[repr(i16)]
```

Comment:
any reason this is `i16` instead of `u16`?

Reply:
I used `u16` here because it was being converted to an `i16` before being parsed. I do not think any negative values have been assigned (other than 0xFFFF being reserved) so it should not make a difference either way. Do you think it would be better to use `u16`?

Reply:
I think my original use of `i16` was a mistake. We should use `u16` for consistency.